scout/server/blueprints/variants/controllers.py

```diff
@@ -117,6 +117,17 @@ def _get_group_assessments(store, case_obj, variant_obj):
             if group_case_id == case_obj["_id"]:
                 continue
 
+            # check events to see if variant received assessments in the other case
+            assessment_events = list(
+                store.evaluation_events(
+                    group_case_id,
+                    variant_obj["simple_id"],
+                    variant_obj["variant_type"],
+                    EVALUATION_VERBS,
+                )
+            )
+            if not assessment_events:
```

Comment:
Very nice in general with starting from events. Ideally we might be eyeing a separate collection like the acmg one to hold all types, but in practice events is going to be mostly assessments anyway, so we may as well just index it up and use it as such. 

I get ambivalent about the trade off here in particular here though: you now fetch all eval events once for all displayed variants, check if they are zero then throw them away. This will be faster only really if there are no evaluations at all in the grouped cases? Otherwise it will be somewhat slower always, since it just adds the check. If there are evaluations at all for the cases, the ordinary solution with fetch be in effect. And since number of variants is limited to the page N, it never scales with a constant variants - though linearly with M cases in case groups.

The way to play this should be to use the event fetch to inform the variant fetch, right, so that we do the cohort variant obj and get_manual_assesment add only for the actual assesment event variants.


In the back of my head there is also a warning chime that there may have been some legacy issue with classifications only on variants and not in events, but we should snooze that one, right? This is the way forward.

Reply:
I agree and in fact this was how I meant to play this feature -> check events and from each event grab the relative variant. Then I had some difficulty with the usual document_id, variant_id and whatnot and it didn't work. I'll try to fix it because this would be a smarter way to do things!